x/feegrant/types/grant_test.go

```diff
@@ -18,57 +19,54 @@ func TestGrant(t *testing.T) {
 	addr2, err := sdk.AccAddressFromBech32("cosmos1p9qh4ldfd6n0qehujsal4k7g0e37kel90rc4ts")
 	require.NoError(t, err)
 	atom := sdk.NewCoins(sdk.NewInt64Coin("atom", 555))
+	ctx := app.BaseApp.NewContext(false, tmproto.Header{})
+	now := ctx.BlockTime()
+	oneYear := now.AddDate(1, 0, 0)
+
 	zeroAtoms := sdk.NewCoins(sdk.NewInt64Coin("atom", 0))
 	cdc := app.AppCodec()
 
 	cases := map[string]struct {
 		granter sdk.AccAddress
 		grantee sdk.AccAddress
-		limit sdk.Coins
-		expires types.ExpiresAt
-		valid bool
+		limit   sdk.Coins
+		expires time.Time
+		valid   bool
 	}{
 		"good": {
 			granter: addr2,
 			grantee: addr,
-			limit: atom,
-			expires: types.ExpiresAtHeight(100),
-			valid: true,
+			limit:   atom,
+			expires: oneYear,
+			valid:   true,
 		},
 		"no grantee": {
 			granter: addr2,
 			grantee: nil,
-			limit: atom,
-			expires: types.ExpiresAtHeight(100),
-			valid: false,
+			limit:   atom,
+			expires: oneYear,
+			valid:   false,
 		},
 		"no granter": {
 			granter: nil,
 			grantee: addr,
-			limit: atom,
-			expires: types.ExpiresAtHeight(100),
-			valid: false,
+			limit:   atom,
+			expires: oneYear,
+			valid:   false,
 		},
 		"self-grant": {
 			granter: addr2,
 			grantee: addr2,
-			limit: atom,
-			expires: types.ExpiresAtHeight(100),
-			valid: false,
-		},
-		"bad height": {
-			granter: addr2,
-			grantee: addr,
-			limit: atom,
-			expires: types.ExpiresAtHeight(-100),
-			valid: false,
```

Comment:
I think we should keep this and change it to "bad time". we can change the expires to: 

```go
expires: now.AddDate(0, -5, 0)
```

Reply:
We cannot set a negative time here, because `blockTime` is `0001-01-01`. And there is no validation for the expiration time in `ValidateBasic`.

Reply:
I was able to set the blocktime in the ctx:

```go
ctx := app.BaseApp.NewContext(false, tmproto.Header{
	Time: time.Now(),
})
```
tests still ran good. You are right though - there is no validation for that in `ValidateBasic`. Maybe we should add this in a separate issue? Wdyt?

Reply:
To do this we have to pass `ctx.BlockTime()` to `ValidateBasic`. I think `ValidateBasic` job is to do stateless checks. Wdyt @aaronc?

Reply:
Maybe we can just use time.Now() for validate basic? its just to make sure user does not put a time that has already passed

Reply:
updated

Reply:
looks good - can we add the test case for it? i dont think the `bad height` test case ever got replaced with the equivalent time version